addons/website_form/static/src/snippets/s_website_form/000.js

```diff
@@ -23,9 +24,13 @@ odoo.define('website_form.s_website_form', function (require) {
             this._super(...arguments);
             this.__started = new Promise(resolve => this.__startResolve = resolve);
         },
-
         start: function () {
             var self = this;
+            if (!this.$target[0].classList.contains('s_website_form_no_recaptcha')) {
+                this._recaptcha = new ReCaptcha(this, {
+                    legalTarget: this.$target[0],
+                });
```

Comment:
Your ReCaptcha init does a lot of async stuff which is not considered here.

You did it with the mixin, it's the same as using a mixin here. In the init: do `new ReCaptcha`, in the willStart: do `this._recaptcha.waitForLibs()`, in the start do `this._recaptcha.addLegalTerms(...)`.

The advantage over a mixin is that you only use what you need. If your recaptcha object had a private method "_doStuff", you won't receive it here and would not use it. Where you would be force to have it with a mixin.

It's all about using a hammer to drive a nail instead of becoming the hammer ;)

Reply:
@qsm-odoo I have pushed some adjustments according to your comment but I am not sure to be interpreting it correctly. Especially for the case of not having the public key and thus not loading any libs. Here I just return false If the methods requiring libs are called and the `waitForLibs` method return false so the widget know it can't use reCaptcha. Also in the widget using the captcha I make sure the libs are loaded before starting the widget as opposed to before where the widget will start and the methods would wait for the libs to load. Is it better to have the form initialization wait for the reCaptcha lib to load or the submit button wait for it. Maybe it does not matter I'm just wondering.